communication/src/main/java/org/phenotips/messaging/internal/DefaultActionManager.java

```diff
@@ -126,7 +125,8 @@ public int sendInitialMails(Connection connection, Map<String, Object> options)
             mail.setTextPart((String) options.get(OPTIONS_MESSAGE_FIELD));
             mail.setSubject((String) options.get(SUBJECT_FIELD_STRING));
             mailsender.sendMail(mail, context);
-            setNotified((String) options.get("patientId"), (String) options.get("matchId"));
+            setNotified((String) options.get("patientId"), this.convertServerId((String) options.get("serverId")),
+                    this.convertServerId((String) options.get("matchId")), (String) options.get("matchServerId"));
```

Comment:
This looks wrong to me. Shouldn't `convertServerId` apply to `matchServerId` instead of `matchId`?